src/compiler/checker.ts

```diff
@@ -13041,6 +13041,16 @@ namespace ts {
                                     // JsxAttributes has an object-literal flag and undergo same type-assignablity check as normal object-literal.
                                     // However, using an object-literal error message will be very confusing to the users so we give different a message.
                                     // TODO: Spelling suggestions for excess jsx attributes (needs new diagnostic messages)
+
+                                    if (errorNode && isJsxOpeningLikeElement(errorNode.parent)) {
+                                        const attributes = errorNode.parent.attributes;
+                                        for (const jsxProperty of attributes.properties) {
+                                            if (jsxProperty.kind === SyntaxKind.JsxAttribute && jsxProperty.name.escapedText === prop.escapedName) {
+                                                // Move the error node to the actual JSX property, instead of pointing to the identifier in the JSX element.
+                                                errorNode = jsxProperty;

```

Comment:
Instead of the entire attribute, can you error only on the attribute name?

Reply:
Yeah, that makes more sense, since it avoids covering up any type errors inside the attribute expression.